independent-projects/qute/core/pom.xml

```diff
@@ -32,6 +32,10 @@
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.vertx</groupId>
+            <artifactId>vertx-core</artifactId>
+        </dependency>
```

Comment:
@mkouba is adding vertx-core dep alright?

Reply:
Not at all ;-)

@mcruzdev this should go in the runtime module of the `quarkus-qute` extension. Ideally, the runtime module should have an optional dependency on `io.quarkus:quarkus-vertx` and the value resolver would be only registered if `io.quarkus.deployment.Capability.VERTX` is present. The build step should be similar to the first version of Cache integration: https://github.com/quarkusio/quarkus/blob/d4c8ba388c40149e421b56c68126dfce8644ee28/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/CacheProcessor.java.

You can make use of the [`@EngineConfiguration` annotation](https://quarkus.io/guides/qute-reference#engine-customization) so that the value resolver is discovered automatically. There is a test that shows an annotated `ValueResolver` in action: https://github.com/quarkusio/quarkus/blob/main/extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/engineconfigurations/resolver/CustomResolversTest.java.

Ping me on zulip if you have more questions...